### PR TITLE
[CI/CD] Add Cypress e2e workflow for gantt chart

### PR DESCRIPTION
**Pull request by RyanL1997**
### Description
Add Cypress e2e workflow for gantt chart

### Related Issue
  * Relate  #296 
 
### Cypress error matches the expected error I captured on: https://github.com/opensearch-project/dashboards-visualizations/pull/296
```
The response we got was:
Status: 406 - Not Acceptable
Headers: {
  "content-type": "application/json; charset=utf-8",
  "osd-name": "fv-az985-504",
  "cache-control": "private, no-cache, no-store, must-revalidate",
  "vary": "accept-encoding",
  "content-encoding": "gzip",
  "date": "Wed, 13 Dec 2023 23:40:25 GMT",
  "connection": "keep-alive",
  "keep-alive": "timeout=120",
  "transfer-encoding": "chunked"
}
Body: {
  "error": "Content-Type header [application/x-www-form-urlencoded] is not supported",
  "status": 406
}
```
Link: https://github.com/opensearch-project/dashboards-visualizations/actions/runs/7202310509/job/19620185810?pr=298#step:20:115

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/cypress-e2e-gantt-chart-test.yml [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+          echo "Start checking OpenSearch Dashboards."
+          for i in {1..60}; do
+            if grep -q "bundles compiled successfully after" "dashboard.log"; then
+              echo "OpenSearch Dashboards compiled successfully."
+              break
+            fi
+            if [ $i -eq 60 ]; then
+              echo "Timeout for 600 seconds reached. OpenSearch Dashboards did not finish compiling."
+              exit 1
+            fi
```
Comment (ps48) [inline .github/workflows/cypress-e2e-gantt-chart-test.yml:125]: Nice!

**Review by ps48 — DISMISSED** [review]

**Comment by ps48** [comment]
Let's backport this. During backport we'll have to modify the below lines:
```
OPENSEARCH_DASHBOARDS_VERSION: 'main'
OPENSEARCH_VERSION: '3.0.0'
OPENSEARCH_PLUGIN_VERSION: '3.0.0.0'
  ```

File: .github/workflows/cypress-e2e-gantt-chart-test.yml [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+          repository: opensearch-project/Opensearch-Dashboards
+          ref: ${{ env.OPENSEARCH_DASHBOARDS_VERSION }}
+          path: OpenSearch-Dashboards
+      
+      - name: Show OpenSearch Logs
+        if: always()
+        run: cat ./opensearch-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/logs/opensearch.log
+        shell: bash
+      
+      - name: Checkout OpenSearch Dashboards
```
Comment (mengweieric) [inline .github/workflows/cypress-e2e-gantt-chart-test.yml:66]: Do we need to checkout Checkout OpenSearch Dashboards twice here?

**Review by mengweieric — DISMISSED** [review]

File: .github/workflows/cypress-e2e-gantt-chart-test.yml (hunk 1 above)
Comment (joshuali925) [reply .github/workflows/cypress-e2e-gantt-chart-test.yml:125]: nice. you can probably simplify this
```bash
if timeout 600 tail -f dashboard.log | grep -q 'bundles compiled successfully after'; then
  # compiled
else
  # timed out
  exit 1
fi
```

File: .github/workflows/cypress-e2e-gantt-chart-test.yml (hunk 2 above)
Comment (mengweieric) [inline .github/workflows/cypress-e2e-gantt-chart-test.yml:66]: do we need Checkout OpenSearch Dashboards twice here?

Comment (RyanL1997) [reply .github/workflows/cypress-e2e-gantt-chart-test.yml:59]: Good catch.. I just removed the first one, and that was just a debugging stuff.

File: .github/workflows/cypress-e2e-gantt-chart-test.yml (hunk 1 above)
Comment (RyanL1997) [reply .github/workflows/cypress-e2e-gantt-chart-test.yml]: @joshuali925 Thanks for the input. Yes, I have applied this. The streaming is more clear than the looping. Nice suggestion.

**Review by joshuali925 — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Review by mengweieric — APPROVED** [review]
Let's gooooo

**Comment by ps48** [comment]
cypress is expected to fail for now. Synced with @RyanL1997 offline, he will create a backport for this to 2.x

**Pull request merged**